src/components/GoogleMapContainer.ts

```diff
@@ -222,6 +237,61 @@ class GoogleMapContainer extends Component<GoogleMapContainerProps, { alertMessa
             ? getStaticMarkerUrl(image.enumImage as string, this.props.defaultMakerIcon)
             : "";
     }
+
+    public static validateProps(props: GoogleMapContainerProps): string {
+        let errorMessage = "";
+        if (props.onClickEvent === "callMicroflow" && !props.onClickMicroflow) {
+            errorMessage = "A 'Microflow' is required for 'Markers' 'Call a microflow'";
+        } else if (props.onClickEvent === "callNanoflow" && !props.onClickNanoflow.nanoflow) {
+            errorMessage = "A 'Nanoflow' is required for 'Markers' 'Call a nanoflow'";
+        } else if (props.onClickEvent === "showPage" && !props.page) {
+            errorMessage = "A 'Page' is required for 'Markers' 'Show a page'";
+        }
+        if (errorMessage) {
+            errorMessage = `Error in google maps configuration: ${errorMessage}`;
+        }
+
+        return errorMessage;
+    }
+
+    private executeOnClickAction = (data: LatLng) => {
```

Comment:
What does this function do creates an object 'and' also executes an action, Function should do one thing. separate this => createOnclickObject , executeOnClick

Reply:
Done